tinkerbell/boots/templates/deployment.yaml

```diff
@@ -29,26 +33,42 @@ spec:
       containers:
         - image: {{ .Values.image }}
           imagePullPolicy: {{ .Values.imagePullPolicy }}
-          {{- if .Values.args }}
-          args: 
-          {{- range .Values.args }}
+          args:
+            - -dhcp-addr={{ printf "%v:%v" .Values.dhcp.ip .Values.dhcp.port }}
+            - -ipxe-tftp-addr={{ printf "%v:%v" .Values.tftp.ip .Values.tftp.port }}
+            - -http-addr={{ printf "%v:%v" .Values.http.ip .Values.http.port }}
+            - -syslog-addr={{ printf "%v:%v" .Values.syslog.ip .Values.syslog.port }}
+          {{- range .Values.additionalArgs }}
             - {{ . }}
           {{- end }}
-          {{- end }}
           env:
             - name: TRUSTED_PROXIES
-              value: {{ required "missing trustedProxies" .Values.trustedProxies | quote }}
-            {{- range $i, $env := .Values.env }}
-            - name: {{ $env.name | quote }}
-              value: {{ $env.value | quote }}
+              value: {{ required "missing trustedProxies" ( join "," .Values.trustedProxies | quote ) }}
+            - name: DATA_MODEL_VERSION
+              value: "kubernetes"
+            - name: FACILITY_CODE
+              value: "lab1"
+            - name: MIRROR_BASE_URL
+              value: {{ printf "%v://%v:%v" .Values.osieBase.protocol $osieBaseIp .Values.osieBase.port | quote }}
+            - name: PUBLIC_IP
+              value: {{ $remoteDhcpIp | quote }}
+            - name: PUBLIC_SYSLOG_FQDN
+              value: {{ $remoteSyslogIp | quote }}
+            - name: TINKERBELL_GRPC_AUTHORITY
+              value: {{ printf "%s:%v" $tinkServerIp .Values.tinkServer.port | quote }}
+            - name: TINKERBELL_TLS
+              value: {{ .Values.tinkServer.tls | quote }}
+            - name: BOOTS_EXTRA_KERNEL_ARGS
+              value: {{ join " " ( append .Values.additionlKernelArgs ( printf "tink_worker_image=%s" ( required "missing tinkWorkerImage" .Values.tinkWorkerImage ) ) ) | quote }}
+            - name: BOOTS_LOG_LEVEL
+              value: {{ .Values.logLevel | quote }}
```

Comment:
we'll need to add this
```yaml
- name: BOOTS_OSIE_PATH_OVERRIDE
  value: {{ printf "%v://%v:%v" .Values.osieBase.protocol $osieBaseIp .Values.osieBase.port | quote }}
```

Reply:
Thanks. Added it in.

Reply:
`BOOTS_OSIE_PATH_OVERRIDE` didn't seem to work for me. I used the `-osie-path-override` flag instead, is that what you meant?